### PR TITLE
docs:add what is RESTful API blog

### PR DESCRIPTION
**Pull request by KID-G**
Changes:
Add what is RESTful API blog

### CODE REVIEW

File: blog/zh/blog/2023/02/08/what-is-restful-api.md [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+|Client–server architecture|通过将用户界面问题与数据存储问题分开，提高了跨多个平台的用户界面的可移植性，并通过简化服务器组件提高了可扩展性。|1. 客户端、服务端解耦。<br>2. 提升用户平台跨平台的可移植性。<br>3. 提升服务器端的可拓展性。|
+|Statelessness|客户端的每个请求到服务器必须包含请求所需的所有信息，并且不能利用服务器上存储的任何上下文，会话状态完全保存在客户端。|1. 易扩展，无会话依赖，任何服务器可以处理任何请求。<br>2. 易缓存，提升程序性能。|
+|Cacheability|要求请求响应中的数据被隐式或显式标记为可缓存或不可缓存。如果一个响应是可缓存的，那么客户端缓存就被授予为以后的等效请求重用该响应数据的权利。|1. 减少带宽。<br>2. 减少延迟。<br>3. 减少服务器负载。<br>4. 隐藏网络状态。|
+|Layered system|通过约束组件行为允许架构由分层层组成，这样每个组件都不能“看到”超出它们与之交互的直接层。通过将系统知识限制在单个层，降低了整个系统的复杂性并促进了底层独立性。|1. 降低整个系统的复杂性。<br>2. 促进底层的独立性。<br>3. 可方便的实施负载均衡。<br>4. 可将业务逻辑和安全策略解耦。|
+|Code on demand (optional)|允许通过下载和执行小程序或脚本形式的代码来扩展客户端功能。|1. 提高系统的可扩展性。|
+|Uniform interface|主要包含四点：<br>1. Resource identification in requests.<br>客户能够通过请求中包含的 URI 来识别一个资源，将服务端资源和客户端请求资源解耦。<br>2. Resource manipulation through representations.<br>当客户端拥有一个资源的表示，如 URI，那么就有足够的信息来修改或者删除资源。<br>3. Self-descriptive messages.<br>每个消息都包括足够的信息来告知客户客户端该如何处理该信息。<br>4. Hypermedia as the engine of application state ([HATEOAS](https://github.com/api7/contents/blob/691305a1293991d7b0b39a0ac9a7a655c935778a/blog/https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FHATEOAS)).<br>客户端不需要任何额外的编码通过服务端返回的资源链接，就可以使得用户获取所有的资源。|1. 简化了整体系统架构。<br>2. 提高了交互的可见性。|
```
Comment (SkyeYoung) [inline blog/zh/blog/2023/02/08/what-is-restful-api.md]:
```suggestion
|约束条件|详述|优势|
|:----|:----|:----|
|Client–server architecture|通过将用户界面问题与数据存储问题分开，提高了跨多个平台的用户界面的可移植性，并通过简化服务器组件提高了可扩展性。|1. 客户端、服务端解耦。<br>2. 提升用户平台跨平台的可移植性。<br>3. 提升服务器端的可拓展性。|
|Statelessness|客户端的每个请求到服务器必须包含请求所需的所有信息，并且不能利用服务器上存储的任何上下文，会话状态完全保存在客户端。|1. 易扩展，无会话依赖，任何服务器可以处理任何请求。<br>2. 易缓存，提升程序性能。|
|Cacheability|要求请求响应中的数据被隐式或显式标记为可缓存或不可缓存。如果一个响应是可缓存的，那么客户端缓存就被授予为以后的等效请求重用该响应数据的权利。|1. 减少带宽。<br>2. 减少延迟。<br>3. 减少服务器负载。<br>4. 隐藏网络状态。|
|Layered system|通过约束组件行为允许架构由分层层组成，这样每个组件都不能“看到”超出它们与之交互的直接层。通过将系统知识限制在单个层，降低了整个系统的复杂性并促进了底层独立性。|1. 降低整个系统的复杂性。<br>2. 促进底层的独立性。<br>3. 可方便的实施负载均衡。<br>4. 可将业务逻辑和安全策略解耦。|
|Code on demand (optional)|允许通过下载和执行小程序或脚本形式的代码来扩展客户端功能。|1. 提高系统的可扩展性。|
|Uniform interface|主要包含四点：<br>1. Resource identification in requests.<br>客户能够通过请求中包含的 URI 来识别一个资源，将服务端资源和客户端请求资源解耦。<br>2. Resource manipulation through representations.<br>当客户端拥有一个资源的表示，如 URI，那么就有足够的信息来修改或者删除资源。<br>3. Self-descriptive messages.<br>每个消息都包括足够的信息来告知客户客户端该如何处理该信息。<br>4. Hypermedia as the engine of application state ([HATEOAS](https://github.com/api7/contents/blob/691305a1293991d7b0b39a0ac9a7a655c935778a/blog/https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FHATEOAS)).<br>客户端不需要任何额外的编码通过服务端返回的资源链接，就可以使得用户获取所有的资源。|1. 简化了整体系统架构。<br>2. 提高了交互的可见性。|
```

This part should be reformated, and `<br>` -> `<br/>`

File: blog/zh/blog/2023/02/08/what-is-restful-api.md [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+## RESTful API 最佳实践
+
+强调组件间的 **统一接口** 是 REST 架构风格与其他基于网络的风格区分开来的核心特征，基于此特征，本文梳理了RSETful 最佳实践，以帮助你更好的设计 API。
+
+### 路径名称避免动词
+
+使用 HTTP 方法来表达资源操作行为，而不是将行为动词定义到路径中。
+
+```shell
+// Good
```
Comment (SkyeYoung) [inline blog/zh/blog/2023/02/08/what-is-restful-api.md]:
```suggestion
# Good
```

In shell, `//` should be `#`

File: blog/zh/blog/2023/02/08/what-is-restful-api.md [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+### 善用 HTTP 状态码
+
+HTTP 标准定义个状态码，大致可以分为以下几类：
+
+|状态码|含义|
+|:----|:----|
+|2xx|成功，操作被成功接收并处理|
+|3xx|重定向，需要进一步的操作以完成请求|
+|4xx|客户端错误，请求包含语法错误或无法完成请求|
+|5xx|服务器错误，服务器在处理请求的过程中发生了错误|
```
Comment (SkyeYoung) [inline blog/zh/blog/2023/02/08/what-is-restful-api.md:136]: ditto

File: blog/zh/blog/2023/02/08/what-is-restful-api.md [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+curl  http://httpbin.org/orders -H "custom-version: v2"
+```
+
+## APISIX 如何助力 RESTful API
+
+作为一个动态、实时、高性能的 API 网关，Apache APISIX 可以在任何 RESTful API 服务上运行，并使用插件来添加新的服务和扩展其功能，这符合 RESTful 定义中的 **Layered system**。此外，对于一些没有遵循 RESTful API 定义的历史服务， APISIX 也可以帮你在**不改动原有业务代码**的情况下完成接口的转换，使你的接口完成 **Uniform interface** 这一 REST 限制条件，使你的 API 可以更好的遵守 RESTful API 规范。
+
+### 分层系统：支持业务逻辑和安全逻辑的分割
+
+你可以只用关注业务逻辑的实现，接口的安全逻辑可以交给 APISIX Authentication 类插件处理，例如 [key-auth](https://apisix.apache.org/docs/apisix/plugins/key-auth)。APISIX 支持大量的 Authentication 插件，我们以 [openid-connet](https://apisix.apache.org/docs/apisix/plugins/openid-connect/)为例如下图所示：
```
Comment (SkyeYoung) [inline blog/zh/blog/2023/02/08/what-is-restful-api.md]:
```suggestion
你可以只用关注业务逻辑的实现，接口的安全逻辑可以交给 APISIX Authentication 类插件处理，例如 [key-auth](https://apisix.apache.org/docs/apisix/plugins/key-auth)。APISIX 支持大量的 Authentication 插件，我们以 [openid-connet](https://apisix.apache.org/docs/apisix/plugins/openid-connect/) 为例如下图所示：
```

File: blog/zh/blog/2023/02/08/what-is-restful-api.md [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+    "upstream": {
+        "type": "roundrobin",
+        "nodes": {
+            "127.0.0.1:80": 1
+        }
+    }
+}'
+```
+
+例如，这个例子表示将请求 /orders 路径的 API 中响应为 200 的状态的请求修改为 201。
```
Comment (SkyeYoung) [inline blog/zh/blog/2023/02/08/what-is-restful-api.md]:
```suggestion
例如，这个例子表示将请求 `/orders` 路径的 API 中响应为 `200` 的状态的请求修改为 `201`。
```

**Review by SkyeYoung — CHANGES_REQUESTED** [review]

File: blog/zh/blog/2023/02/08/what-is-restful-api.md (hunk 3 above)
Comment (KID-G) [reply blog/zh/blog/2023/02/08/what-is-restful-api.md:137]: I don’t know what‘s wrong with this table

Comment (KID-G) [reply blog/zh/blog/2023/02/08/what-is-restful-api.md:136]: > ditto
I don’t know what‘s wrong with this table

Comment (SkyeYoung) [reply blog/zh/blog/2023/02/08/what-is-restful-api.md:136]: Just format it, for Format it to provide a better reading experience.

If no, I think it is okay 🤔

File: blog/zh/blog/2023/02/08/what-is-restful-api.md [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+    title: "Author"
+    url: "https://github.com/litesun"
+    image_url: "https://github.com/litesun.png"
+keywords: 
+- RESTful
+- Apache APISIX
+- API
+description: 本文将介绍什么是 RESTful API 以及我们如何使用它。
+
+tags: [Ecosystem]
```
Comment (guoqqqi) [inline blog/zh/blog/2023/02/08/what-is-restful-api.md]:
```suggestion
description: 本文将介绍什么是 RESTful API 以及我们如何使用它。
tags: [Ecosystem]
```

**Review by guoqqqi — DISMISSED** [review]

File: blog/zh/blog/2023/02/08/what-is-restful-api.md [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+        }
+    }
+}'
+```
+
+你也可以同样使用该插件进行 API 版本管理上的操作。
+
+#### 使用 [response-rewrite](https://apisix.apache.org/docs/apisix/plugins/response-rewrite/) 插件改写服务端响应
+
+当我们的历史 API 存在响应状态码不规范时，我们可以通过 response-rewrite 代理 response 响应从而达到修改响应状态码的目的。
```
Comment (SkyeYoung) [inline blog/zh/blog/2023/02/08/what-is-restful-api.md]:
```suggestion
当我们的历史 API 存在响应状态码不规范时，我们可以通过 `response-rewrite` 代理 response 响应从而达到修改响应状态码的目的。
```

**Review by SkyeYoung — CHANGES_REQUESTED** [review]

**Review by SkyeYoung — APPROVED** [review]

**Review by guoqqqi — APPROVED** [review]

**Pull request merged**